numpy/ma/tests/test_core.py

```diff
@@ -4981,6 +4981,10 @@ class Sub(type(np.ma.masked)): pass
         assert_(a is not np.ma.masked)
         assert_not_equal(repr(a), 'masked')
 
+    def test_attributes_readonly(self):
+        assert_raises(AttributeError, setattr, np.ma.masked, 'shape', (1,))
+        assert_raises(AttributeError, setattr, np.ma.masked, 'dtype', np.int64)
```

Comment:
Previous commit had `MaskError`, but this seems more useful for code that does:

```
try:
    x.shape = (1, 2)
except AttributeError:
    x = array(x)
    x.shape = (1, 2)
```

Reply:
Yes, better.